### PR TITLE
Update go and fyne versions

### PR DESCRIPTION
**Pull request by russellsteadman**
### Description:

Updates Docker base to Go v1.18.2, Fyne v2.1.4

### Checklist:

Note: `develop` branch was already testing on Go v1.18.X in `ci.yml`

- [x] Tests included.
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.

### CODE REVIEW

**Comment by Jacalz** [comment]
Thanks for the PR. I think we were talking about getting Go 1.18 rolled out together with some other larger work but I’m not sure. Cc @lucor.

**Comment by lucor** [comment]
Yes, the plan was to release a new version of fyne-cross with Go 1.18.x when fyne 2.2.0 is out.

**Comment by andydotxyz** [comment]
It is time :) Plus an update with some important fixes followed, so this should now use Fyne `v2.2.1`

File: docker/base/Dockerfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
-ARG GO_VERSION=1.17.8
+ARG GO_VERSION=1.18.2
 # fyne stable branch
-ARG FYNE_VERSION=v2.1.3
+ARG FYNE_VERSION=v2.1.4
```
Comment (lucor) [inline docker/base/Dockerfile:3]:
```suggestion
ARG FYNE_VERSION=v2.2.1
```

**Review by lucor — CHANGES_REQUESTED** [review]
As @andydotxyz pointed out into the comment the new version of Fyne is out :)
If you have any chance to update your PR using the latest versions of Fyne CLI and Go, we should be ready to go

**Comment by russellsteadman** [comment]
@lucor Updated per suggestions

**Review by lucor — APPROVED** [review]
Thanks

**Pull request merged**